### PR TITLE
Fix errors that can occur during shape initialization

### PR DESCRIPTION
**Pull request by InfiniteLee**
Add some better checks during shape initialization. Wait until body is loaded before adding shape.

### CODE REVIEW

**Comment by netpro2k** [comment]
lgtm

File: src/components/body/body.js [hunk 1]
```diff
@@ -104,6 +104,10 @@ var Body = {
       return;
     }
 
+    if (!this.body) {
+      console.warn('body is not loaded');
```
Comment (donmccurdy) [inline src/components/body/body.js]: Could you rephrase as `shape cannot be added before body is loaded`?

File: src/components/shape/shape.js [hunk 2]
```diff
@@ -86,7 +91,14 @@ var Shape = {
         return;
     }
 
-    this.bodyEl.components[bodyType].addShape(shape, offset, orientation);
+    var bodyComponent = this.bodyEl.components[bodyType];
```
Comment (donmccurdy) [inline src/components/shape/shape.js]: I'd be more comfortable if we could add `dependencies: ['body']` to this component, otherwise this feels either could technically initialize first. How important is the behavior above where the `body` component could be on an ancestor of the shape, rather than the same element?

I guess this also implies that `dynamic-body` and `static-body` are incompatible with `shape`, which TBH I'm OK with. We could also make them helpers that add the `body` component, for backward-compat, rather than aliases.

Comment (InfiniteLee) [reply src/components/shape/shape.js]: Right now, Hubs relies on this functionality because of how we inflate components defined in extras data on dynamically loaded gltf models. It would take some work to change it in hubs to nor rely on the `body` being on an ancestor (and I'm not entirely sure how soon we can do it) but I agree it would be cleaner to simply require the `body` component. I am also in favor of making `dynamic-body` and `static-body` into helpers. May I suggest we make those changes in a future PR though?

Comment (donmccurdy) [reply src/components/shape/shape.js]: That's fine, I'd just move `var bodyComponent = this.bodyEl.components[bodyType];` inside the `body-loaded` event callback in case the shape component initializes before the body component.

**Comment by donmccurdy** [comment]
Thanks!
